rclpy/test/test_logging_rosout.py

```diff
@@ -17,20 +17,23 @@
 from rcl_interfaces.msg import Log
 import rclpy
 from rclpy.executors import SingleThreadedExecutor
+from rclpy.logging import LoggingSeverity
 
 TEST_PARAMETERS = [
+    # name, enable_global_rosout_logs, enable_node_rosout, expected_data
     ('enable_global_rosout_enable_node_rosout', True, True, True),
     ('enable_global_rosout_disable_node_rosout', True, False, False),
     ('disable_global_rosout_enable_node_rosout', False, True, False),
     ('disable_global_rosout_disable_node_rosout', False, False, False),
 ]
 
-raw_subscription_msg = None  # None=No result yet
+receive_expected_rosout_msg = False  # True if expected rosout msg is received
```

Comment:
Why did you end up changing from storing the message to storing a boolean?  It doesn't matter a *huge* amount here, but storing the message meant we could do more detailed checks on it (like it contained bytes, and the length wasn't 0).  Unless there is a good reason, I'd suggest to go back to storing the message.

Reply:
i guess this is because of my suggested code, but i do agree with you.

Reply:
Thanks for comments.

OK. I want to store the message but not raw message. Check the message outside the callback.